### DISCUSSION TITLE
Quasar and Pinia

### DISCUSSION THREAD

**Discussion by MilosPaunovic** [quasarframework/quasar#11882] [Ideas / Proposals] [answered]
Hello guys,

I've been thinking about something and decided to ask a few questions about it.

Since `pinia`, as from recently, is under `vue` on Github ([tweet](https://twitter.com/posva/status/1471381656351223811?s=20), [repo](https://github.com/vuejs/pinia)), maybe it's time to make an integration for it in Quasar? Truth be told, it's still a massive advantage for `vuex` in terms of weekly npm downloads (900K > 20K), but I believe that more and more developers will accept `pinia` now as it is integral part of `Vue` ecosystem.

What I've been thinking is a change of Quasar CLI, which will ask the user for a state management option during new project creation, and if the answer is yes, suboptions would include either `vuex` or `pinia`. That would require changes to both `@quasar/app` and `quasar-stater-template` repos, as well as new examples in the docs.

My questions are following:
1. Is anyone already working on it?
2. Are the members of the core team interested in something like this?
3. If yes, should I focus only on `Q2`?
4. Should I develop both `.js` and `.ts` options, in parallel?

**Comment by yusufkandemir** [comment, marked as answer]
It's released in [@quasar/app-vite-v1.0.0-beta.1](https://github.com/quasarframework/quasar/releases/tag/%40quasar%2Fapp-vite-v1.0.0-beta.1) and [@quasar/app-webpack-v3.4.0](https://github.com/quasarframework/quasar/releases/tag/%40quasar%2Fapp-v3.4.0).
~~It's implemented in #12707, followed by #12708 and #12742. We will add the documentation for it. It will be included in the next release of Quasar CLI(both Webpack and Vite-based 💪).~~
We offer:
- Project scaffolding for both JS and TS
- Support SSR hydration
- `quasar new store whatever` variant for Pinia too:
  - It adds Pinia to projects that got created without Pinia at first.
  - It creates new stores (`src/stores/<store-name>.[js|ts]`)
